homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -1014,15 +1022,65 @@ async def recognize_intent(
         )
 
         try:
-            conversation_result = await conversation.async_converse(
-                hass=self.hass,
+            user_input = conversation.ConversationInput(
                 text=intent_input,
+                context=self.context,
                 conversation_id=conversation_id,
                 device_id=device_id,
-                context=self.context,
                 language=self.pipeline.conversation_language,
                 agent_id=self.intent_agent,
             )
+            if user_input.language == MATCH_ALL:
+                # We only load local intents for one language
+                user_input.language = (
+                    self.pipeline.stt_language
+                    or self.pipeline.tts_language
+                    or self.hass.config.language
+                )
+
+            # Sentence triggers override conversation agent
+            if (
+                trigger_response_text
+                := await conversation.async_handle_sentence_triggers(
+                    self.hass, user_input
+                )
+            ):
```

Comment:
Hmm, I'm not sure I really see a use case for when I would want my sentence triggers to work and the intents to not work. I either want a pipeline with all local processing enabled, or all disabled, never something in between.
Also the distinction is even more blurry, because users not only create sentence triggers, they also create custom intents as well.
I would therefore recommended keeping all local processing under one setting.

Reply:
I could see using an LLM agent, but then filling in some gap that the LLM doesn't have visibility to with a custom sentence. I agree with @synesthesiam that it seems prudent to act on what the user is explicitly saying. They are saying they want a sentence to be handled via their automation configuration, and they are explicitly setting the conversation agent. They are not explicitly choosing the default matched sentences.

The proposed option here is basically one that allows opting _out_ of the default sentences rather than opting out of all local matching.

Reply:
>The proposed option here is basically one that allows opting out of the default sentences rather than opting out of all local matching.

No, that is not what this option does. It opts out of all sentence matching with intents (both default and user-defined), and does not opt out of matching with automations.

>They are saying they want a sentence to be handled via their automation configuration

But what if they want some assist pipelines to be LLM + local matching, but some pipelines LLM only? With the way it is currently, that would not be possible.

>the user is explicitly saying

The user literally nowhere does say to handle stuff with automations when defining an LLM conversation pipeline, which is precisely the problem.